tools/ci/src/main.rs

```diff
@@ -88,12 +88,24 @@ fn main() {
     }
 
     if what_to_run.contains(Check::COMPILE_FAIL) {
-        // Run UI tests (they do not get executed with the workspace tests)
-        // - See crates/bevy_ecs_compile_fail_tests/README.md
-        let _subdir = sh.push_dir("crates/bevy_ecs_compile_fail_tests");
-        cmd!(sh, "cargo test --target-dir ../../target")
-            .run()
-            .expect("Compiler errors of the ECS compile fail tests seem to be different than expected! Check locally and compare rust versions.");
+        {
+            // ECS Compile Fail Tests
+            // Run UI tests (they do not get executed with the workspace tests)
+            // - See crates/bevy_ecs_compile_fail_tests/README.md
+            let _dir_raii = sh.push_dir("crates/bevy_ecs_compile_fail_tests");
```

Comment:
not a fan of the new variable name

Reply:
I chose to rename it because I wasn't aware that `_subdir` is actually a RAII that restores the previous working directory when dropped. I can rename it back, though!